examples/statistics/confidence_ellipse.py

```diff
@@ -0,0 +1,259 @@
+#
+# This function plots the confidence ellipse of the covariance of the given
+# array-like variables x and y. The ellipse is plotted into the given
+# axes-object ax.
+#
+# The radiuses of the ellipse can be controlled by n_std which is the number
+# of standard deviations. The default value is 3 which makes the ellipse
+# enclose 99.7% of the points (given the data is normally distributed
+# like in these examples).
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.patches import Ellipse
+import matplotlib.transforms as transforms
+
+
+def confidence_ellipse(x, y, ax, n_std=3.0, **kwargs):
+    """
+    Create a plot of the covariance confidence ellipse of `x` and `y`
+
+    Parameters
+    ----------
+    x, y : array_like, shape (n, )
+        Input data
+
+    ax : matplotlib.axes object to the ellipse into
+
+    n_std : number of standard deviations to determine the ellipse's radiuses
+
+    Returns
+    -------
+    None
+
+    Other parameters
+    ----------------
+    kwargs : `~matplotlib.patches.Patch` properties
+    """
+    if x.size != y.size:
+        raise ValueError("x and y must be the same size")
+
+    cov = np.cov(x, y)
+    pearson = cov[0, 1]/np.sqrt(cov[0, 0] * cov[1, 1])
+    # Using a special case to obtain the eigenvalues of this
+    # two-dimensionl dataset.
+    ell_radius_x = np.sqrt(1 + pearson)
+    ell_radius_y = np.sqrt(1 - pearson)
+    ellipse = Ellipse((0, 0),
+        width=ell_radius_x * 2,
+        height=ell_radius_y * 2,
+        **kwargs)
+
+    # Calculating the stdandard deviation of x from
+    # the squareroot of the variance and multiplying
+    # with the given number of standard deviations.
+    scale_x = np.sqrt(cov[0, 0]) * n_std
+    mean_x = np.mean(x)
+
+    # calculating the stdandard deviation of y ...
+    scale_y = np.sqrt(cov[1, 1]) * n_std
+    mean_y = np.mean(y)
+
+    transf = transforms.Affine2D() \
+        .rotate_deg(45) \
+        .scale(scale_x, scale_y) \
+        .translate(mean_x, mean_y)
+
+    ellipse.set_transform(transf + ax.transData)
+    ax.add_patch(ellipse)
+
+
+#############################################################################
+#
+# A helper function to create a correlated dataset
+# """"""""""""""""""""""""""""""""""""""""""""""""
+#
+# Creates a random two-dimesional dataset with the specified
+# two-dimensional mean (mu) and dimensions (scale).
+# The correlation can be controlled by the param 'dependency',
+# a 2x2 matrix.
+
+def get_correlated_dataset(n, dependency, mu, scale):
+    latent = np.random.randn(n, 2)
+    dependent = latent.dot(dependency)
+    scaled = dependent * scale
+    scaled_with_offset = scaled + mu
+    # return x and y of the new, correlated dataset
+    return scaled_with_offset[:, 0], scaled_with_offset[:, 1]
+
+
+#############################################################################
+#
+# Positive correlation
+# """"""""""""""""""""
+#
+
+fig, ax_pos = plt.subplots(figsize=(6, 6))
```

Comment:
Since the first three plots use the same code and just plot different parameters, you could join them into one figure:

~~~
np.random.seed(0)

PARAMETERS = {
    'Positive correlation': np.array([[0.85, 0.35],
                                      [0.15, -0.65]]),
    'Negative correlation': np.array([[0.9, -0.4],
                                      [0.1, -0.6]]),
    'Weak correlation': np.array([[1, 0],
                                  [0, 1]]),
}
    
mu = np.array([2, 4]).T
scale = np.array([3, 5]).T

fig, axs = plt.subplots(1, 3, figsize=(9, 3))
for ax, (title, dependency) in zip(axs, PARAMETERS.items()):
    x, y = get_correlated_dataset(500, dependency, mu, scale)
    ax.scatter(x, y, s=0.5)
    
    ax.axvline(c='grey', lw=1)
    ax.axhline(c='grey', lw=1)

    confidence_ellipse(x, y, ax, facecolor='none', edgecolor='red')
    
    ax.scatter([mu[0]], [mu[1]], c='red', s=3)
    ax.set_title(title)
~~~

Reply:
Nice one! This one goes in there :-)